protostar/configuration_file/configuration_file_v2.py

```diff
@@ -92,20 +91,24 @@ def get_contract_source_paths(self, contract_name: str) -> list[Path]:
             for path_str in contract_section[contract_name]
         ]
 
-    def get_command_argument(
+    def get_argument_value(
```

Comment:
how is configuration file related to arguments? this class should not be handling that logic

Reply:
`ConfigurationFile` apart from the "normal" interface implements `ArgumentValueProviderProtocol`. 
That interface could be implemented in another class, however it would increase the complexity by introducing a new class and complicating code responsible for class instantiation.

Reply:
I'd had to pick what module depends on which module:
`protostar.cli` -> `protostar.configuration_file` or `protostar.configuration_file` -> `protostar.cli`. I picked the second one because the structure of the composition file is affected by CLI, not vice versa.

Reply:
I think the point of changes made long time ago was to decompose the class `Project` into smaller classes. Creating one big class for everything is nonsense taking it into account and we can consider the time wasted.

Reply:
> I think the point of changes made long time ago was to decompose the class Project into smaller classes. Creating one big class for everything is nonsense taking it into account and we can consider the time wasted.

This class has 64 lines...

> ...decompose the class Project into smaller classes

`ProtostarTOML` sliced the Project class vertically and was a hybrid object. The current approach slices classes horizontally.

Reply:
> This class has 64 lines...
Method, wrapped into the class

> sliced the Project class vertically. The current approach slices classes horizontally.
I do not understand what do you mean by that

Reply:
@Arcticae what is your opinion?

Reply:
- Vertical - by features 
- Horizontal - by technological concepts

Horizontal slicing works better because it makes supporting old and new configuration file at the same time easier.

Reply:
In my opinion this breaks SRP, which is going to result in worse code in terms of maintainability. Can't really tell if the "slicing" alteration is making things better on the whole though, it doesn't ocur to me that the approach was modified in an obvious way

Reply:
> In my opinion this breaks SRP...

[SRP is often misunderstood.](https://dev.to/bourzayq_khalid/srp-is-the-most-misunderstood-principle--2b64)

"Horizontal slicing" doesn't violate SRP. Implementing `ArgumentValueProviderProtocol` methods in this class could be understood as a violation of SRP, however I did on the purpose to avoid creating too many classes.

Reply:
I think it would be easier and simple to create new class than to isolate an abstract protocol.

Reply:
Responded to that in another thread.